packages/transport/src/core/CryptoProviderMapping.ts

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { DeviceBoundKeyHandle, hasProviderForSecurityLevel, PortableKeyHandle } from "@nmshd/crypto";
+import { SecurityLevel } from "@nmshd/rs-crypto-types";
+import { CoreCrypto } from "./CoreCrypto";
+
+export const CryptoProviderTypes = {
+    Software: "Software",
+    Hardware: "Hardware",
+    Network: "Network",
+    LEGACY: "LEGACY"
+} as const;
+
+export const CryptoPurpose = {
+    DeviceKeyPair: "deviceKeyPair",
+    BaseKey: "baseKey",
+    Default: "default"
+} as const;
+type CryptoPurpose = (typeof CryptoPurpose)[keyof typeof CryptoPurpose];
+
+export const CryptoKeyType = {
+    Signature: "signature",
+    Encryption: "encryption",
+    Derivation: "derivation",
+    Exchange: "exchange"
+} as const;
+type CryptoKeyType = (typeof CryptoKeyType)[keyof typeof CryptoKeyType];
+
+export const CryptoObject = {
+    AccountController: "AccountController",
+    AnonymousTokenController: "AnonymousTokenController",
+    Certificate: "Certificate",
+    DeviceController: "DeviceController",
+    DeviceSecretController: "DeviceSecretController",
+    IdentityController: "IdentityController",
+    FileController: "FileController",
+    MessageController: "MessageController",
+    RelationshipTemplateController: "RelationshipTemplateController",
+    RelationshipsController: "RelationshipsController",
+    RelationshipSecretController: "RelationshipSecretController",
+    SecretController: "SecretController",
+    TokenController: "TokenController"
+} as const;
+type CryptoObject = (typeof CryptoObject)[keyof typeof CryptoObject];
+
+export const ALL_CRYPTO_PROVIDERS = ["SoftwareProvider", "AndroidProvider"];
+
+const CRYPTO_OPERATION_OBJECT_MAP: Partial<Record<CryptoObject, CryptoKeyType[]>> = {
```

Comment:
For me, this is a duplicate of OBJECT_OPERATION_PREFERENCES with less details :)